### PR TITLE
Update back links across Referee flow

### PR DESCRIPTION
**Pull request by george-bit**
## Context

As part of the work completed on `3261-references-save-at-end-of-flow`
this back link update should have been included. 

## Changes proposed in this pull request

It will direct candidate back to edit form to prevent duplication of referee.

## Guidance to review

Test in review app.

## Link to Trello card

https://trello.com/c/GuiJ30Jg/3397-catch-all-give-back-links-explicit-targets

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] This code does not rely on the addition/removal of Azure config environment variables in the same Pull Request
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

**Comment by frankieroberto** [comment]
@george-bit this looks good, but there's a couple of other back links which need fixing, it think:

1) If you go to "Add a referee" > "Continue" > choose one, "Continue" > "Back" > "Back" > "Back" > "Back" – then you get stuck in a loop between the Reference type and name pages.

2) If you do the same as above but get as far as the "How do you know this referee..." page, and then use the Back links, you end up in a loop between the name page and that page.

The way I've implemented this in the past is to only add the `?return_to=summary` query string on the "Change" links on the summary pages, and then if that's not present, always link back to the previous page in the flow.

**Review by davidgisbey — APPROVED** [review]

**Comment by george-bit** [comment]
@frankieroberto Think I've fixed it  🤞 

Please have a go in the review app and let me know

**Comment by frankieroberto** [comment]
@george-bit it's getting closer!

Remaining issues:

1) In the Add referee flow, if you do "Add another referee" -> "Continue" -> "Back" then you should be back on the interstitial "Choose your referees" page, but instead end up on the review page.

2) When you do "Add another referee" the interstitial page should have a "Back" link to the review page, at the moment it always has "Back to application". It should still be "Back to the application" if you arrive at this page straight from the application page though via the "Add your references" link (ie when adding the very first reference).

3) From the "Check your answers before sending your request" page, if you click "Change" on the name, email address or relationship rows, and then click "Back", you end up on the References review page instead of the "Check your answers" page. Clicking the Change link for Reference type and then clicking "Back" works correctly though. Do we need to add a `?return_to=review_unsubmitted` query string for the others?

4) Where adding a new referee, if you do "Add referee" -> "Continue" -> select Academic, press "Continue" -> leave name field blank, press "Continue" then you get a "Back" link to the "What is the referees name?" page, rather than the reference type page. If you also keep pressing "Back" you just remain on the same page.

Sorry! This is quite a complicated flow! Give me a shout if you can't repeat any of these, or if we need to take a call on which ones to fix immediately and which to leave.

File: app/views/candidate_interface/references/candidate_name/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<% content_for :title, title_with_error_prefix(t('page_titles.references_candidate_name'), @reference_candidate_name_form.errors.any?) %>
+<% content_for :before_content, govuk_back_link_to(candidate_interface_references_review_path) %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_with(
+      model: @reference_candidate_name_form,
+      url: candidate_interface_references_update_candidate_name_path(@reference.id, return_to: params[:return_to]),
```
Comment (davidgisbey) [inline app/views/candidate_interface/references/candidate_name/edit.html.erb:8]: I don't think we need this view as you are only able to edit this via the personal details section. This endpoint isn't in the routes either.

Comment (george-bit) [reply app/views/candidate_interface/references/candidate_name/edit.html.erb:8]: My bad I think I added this in error thinking it was the 'name' attribute 🤦

**Comment by davidgisbey** [comment]
There are probably more elegant solutions, bit i've done the work to fix all the backlinks. It definitely needs a review though. I've given it a decent manual test, but it would be good if someone else could try and break it too

**Comment by george-bit** [comment]
Looking good David, much better to separate the two flows out,

Tested a manually and is now working really well. Just two minor comments inline:

1. When you go back from the name page in the 'new' flow the parameter for the referee type is not retained. I appreciate the difficulty in passing back the param in this specific circumstance only though.
2. When travelling to the edit pages, submitting the value returns the candidate to the review page if successful. Previously i had some logic implemented here that would redirect the candidate first to any other field which was not filled in and then subsequently to the unsubmitted review page. Appreciate it's simpler in it's current form though.

I don't think either of these area huge problem though and think we could probably merge as is 👍

**Comment by davidgisbey** [comment]
Hey George

1. So i think it does for everything, but school-based as it passes it back as an underscore. Good spot!
2. I might be wrong, but i think generally if you click to edit a field you should always be returned to the page you chose to edit from rather than being directed back into a flow

Wdyt Frankie?

**Review by frankieroberto — APPROVED** [review]
Awesome work - all the back links work as expected, in all contexts, even when there are error messages shown.

The only remaining bug I can find is the one @george-bit mentioned:

> When you go back from the name page in the 'new' flow the parameter for the referee type is not retained.

Is this an easyish fix? I note that the type is in the URL path already?

**Comment by george-bit** [comment]
👍

**Comment by davidgisbey** [comment]
Yea all good. It was just a tiny fix

**Pull request merged**